pkg/kubelet/status/status_manager.go

```diff
@@ -829,32 +837,58 @@ func (m *manager) syncBatch(all bool) int {
 		}
 	}()
 
+	podInfos := make(map[string]v1.Pod)
+	updatedStatusesCount := len(updatedStatuses)
+	if m.nodeName != "" && updatedStatusesCount >= defaultMinListPodCount {
+		podOpts := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("spec.nodeName", m.nodeName).String()}
+		pods, err := m.kubeClient.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), podOpts)
```

Comment:
This can kill us on the control-plane side.

This LIST call would go to etcd, which doesn't understand k8s resource model so would return *every single pod in the cluster* which kube-apiserver then needs to deserialize, filter out non-relevant ones and return the result.
This is extremely expensive call on the control-plane side and definitely NOT something we want.

We can mitigate it by setting ResourceVersion: 0 to have it served from cache, but then we don't have consistent data, which I'm not 100% sure is the right path forward. But if we decide it's not, then this change is no-go from scalability POV.

Reply:
I think that parallel invocation of `syncPod` should be capable of meeting the performance requirements.